### DISCUSSION TITLE
Need Indication On Fader Panel That Someone Has Muted Themselves

### DISCUSSION THREAD

**Discussion by maallyn** [jamulussoftware/jamulus#1025] [UI Related]
Currently there is an indication that a user has muted you. That is the small light above a fader in your panel.

What I am suggesting is to have an indicator above someone's fader that that person has muted themself. This would be good for the sound engineer or the stage manager or the director or the producer to know that someone has muted themself  and their band is about to go onto the studio. 

With such an indicator, someone can message that person (or tell them verbally in the backstage) to please un-mute themselve before going onto the stage.

Currently, the sound engineer would have to ask the stage manager go into the studio, then un-mute the stage manager and then ask the stage manager to ask the musician to un-mute, which is really hard during a live performance and it may result in the band to have to re-start their piece, which can be frustrating to the overall operation/timing to that evening's show.

For matters of screen/panel real estate, could this be done by re-purposing the current mute light that indicates  that that person is muting you and is not hearing you? I would think that it would be more important for me to know if someone is muted themselves rather than muting me as the sound engineer. A performer can mute the sound engineer and it does not make any difference on the show as the sound engineer is not supposed to be heard on the stage.

**Comment by pljones** [comment]
Muting myself in most jams is personal to me.  No one else is interested in it.  It allows me to play along without them having to be aware of me.  That's its purpose.

**Comment by dcorson-ticino-com** [comment]
Just to be sure I am understanding you correctly, you are talking about the "Mute Myself" feature which stops the stream to the server.  You are not talking about the "Mute" feature which means that a musician will not be heard in your personal mix.
Is that correct ?

**Comment by jugo6** [comment]
Its a balance between privacy (pljones) and being helpful to each other. In a good jam or rehearsal or won't hurt to be open on this, i think.

A "muted myself" indicator would also be helpful in my daily choir rehearsals, as i encourage people to mute themselves if they are adjusting headsets, or want to practice silently. But they forget to unmute :-)

**Comment by pljones** [comment]
Another point I thought of.  It requires both clients - the "self-muted" and the "observer" - _and_ the server to support a new protocol indicator.  If any one of those three doesn't provide support, then the information won't be available.  So not seeing that someone is muted is no indication that they are not muted.  It's simply an indication that you don't know they're not muted.

**Comment by maallyn** [comment]
Folks: To answer Don, this is for Mute Myself, not the one that prevents you from hearing someone else. The current little red lights above the faders on your panel mean that that person is muting you. What I am suggesing is another light above somoene's fader that indicates that they are self muting. This, I feel, is important to the sound engineer and the manager. When you, the performer, is called by the manager to go back stage (from the waiting room); the manager can make sure that you have not muted yourself. While you are in the Backstage, the manager can tell you to please unmute yourself before you are called onto the studio stage itself. I had an incident while I was the sound engineer. I had noticed no sound coming from one of the guitarists. I heard no sound and their VU lights next to their fader on the sound engineer's Jamulus client panel were not flashing, which suggested to me that they were muted. I ended up having to ask the manager to go into the sudio, then I had to mute the manager on the sound engineer's console so they would not be heard on the live stream, and then the manager would ask the person to please un-mute. This is very disruptive during a live performance it was nerve racking to me as the sound engineer. I was embarassed and shaking when that performance was over. If there is a visual indication that the person is muted prior to them being in the studio (while they are in the waiting room), it would save the sound engineer some grief as the sound engineer already has enough to contend with. This is all about improving the quality of the show and reducing the un-needed stress level of the sound engineer and the other players.

**Comment by nefarius2001** [comment]
I see it also as a balance of privacy, but useful in groups that trust each other. 

How about: new indicator as proposed, and a checkbox in the settings „do not indicate I am muting my stream“ (default unchecked)

**Comment by gilgongo** [comment]
Incognito mode :-)

**Comment by pljones** [comment]
Mark, you have to realise that the vast majority of users on Jamulus _are not in Worldjam_.  They're just using random servers around the internet and have no desire to have - be default - their status shown.  There are seven days in a week of twenty four hours.  Worldjam has this problem with only its users for two and a half hours.

So I'd also be against it being shown by default.

**Comment by ghost** [comment]
Hi,
I would be in favor of this "I've muted myself" indicator if it is a trivial addition (maybe server can detect it by noticing no audio stream).
I do not think there is much of a privacy concern on the part of the person who has muted themself because their slider gives them away.
However, the others have a privacy concern because they don't know if they are being spied on.

**Comment by pljones** [comment]
"src/client.cpp" lines 1081..1088 are how Mute Myself works.  Basically, rather than encode the buffer of audio read from the input, encode zeros.

Of course, that buffer of audio read could have been zeros.  So the server _currently_ doesn't know.

We could, as we have for the "has me muted" icon, add another client message to the server, which then gets broadcast to everyone else on the server (unlike the "has me muted", which only comes to "me" -- even that is telling me who has me muted, which is actually invading someone else's privacy to a degree - I've not asked them and they've not been told they've told me).

But, as I said, that means
- every client needs upgrading to send and receive the message
- every server needs upgrading to receive and broadcast the message

otherwise you don't know if not seeing the indicator means someone hasn't upgrade or hasn't muted.

**Comment by pljones** [comment]
@DavidSavinkoff 
> I do not think there is much of a privacy concern on the part of the person who has muted themself because their slider gives them away.

I'm not sure what you mean by this.  Can you show it?

**Comment by DominikSchaller** [comment]
I'm also in favor of it, as every chorus director has that problem (is my singer muted, are they just not singing etc.). Actually that happened twice tonight, so would be helpful to be able to give direct feedback to the singer.

I also really don't understand the privacy argument here. What's private about if I press the "mute myself" button or not?

It's definitely less invading than the "has me muted" icon.

Also we shouldn't make the argument that clients or servers need to be updated. That's part of what we do here. You should think more about where Jamulus is in a year or two instead of cancelling feature requests by saying "server needs to be updated, the protocol doesn't allow it".

**Comment by ghost** [comment]
> @DavidSavinkoff
> 
> > I do not think there is much of a privacy concern on the part of the person who has muted themself because their slider gives them away.
> 
> I'm not sure what you mean by this. Can you show it?
>
Every person that has joined a given Jamulus session will have a mixer panel slider associated with them. This lets everyone else know that you are there and satisfies privacy laws of all parties (I would expect - IANAL). Furthermore, someone who is listening without their microphone turned on is not part of a conversation (maybe the mixer panel slider associated with them legally makes them part of the conversation because others can see them). Oh, the privacy (<:

**Comment by houbsta** [comment]
> Mark, you have to realise that the vast majority of users on Jamulus _are not in Worldjam_. They're just using random servers around the internet and have no desire to have - be default - their status shown. There are seven days in a week of twenty four hours. Worldjam has this problem with only its users for two and a half hours.
> 
> So I'd also be against it being shown by default.

Fair point, unless in other contexts this is useful. If somebody hits "mute myself" they can then move away from the UI and be playing along and not be heard by everyone else, without realising it. In choirs, big bands, or many other use cases this can be frustrating. Other members of the same jam cannot tell and advise their friend to unmute themselves, because they could have muted themselves by switching off their mic, using a mute on their interface or in a DAW etc.

From a privacy perspective, you can always mute yourself elsewhere than in Jamulus and everyone would be none the wiser. I don't know that it's a privacy concern.

As far as needing new client and server versions, yes that's a given.

**Comment by DominikSchaller** [comment]
FWIW this is also a feature in Zoom and MS Teams where you can see if
someone else is muted or not. In Zoom you actually see four possible states:
- Not connected with sound
- Muted
- Unmuted
- Connected by phone

Obviously the last one is not needed, but the other are helpful. Unrelated,
but it would be amazing to also see the delay of everybody.

**Comment by pljones** [comment]
> Connected by phone

Ah, the days of dial-up IP.  At least it was 56Kb/s by then.  Watch that OPUS codec cry...

OK, so if it's accepted that:
* only clients that have implemented the new feature
* only when connected to servers that have implemented the new feature
* and only for other connected clients that have implemented the new feature
* and not when muted outside Jamulus
* and not when simply not making any sound

then a solution is possible.

Now, if I already have Mute Myself enabled when I join a server, I assume this needs to immediately tell everyone my status?  And it should then only send updates to status?  (Same way "has me muted" does, I guess.)

**Comment by gene96817** [comment]
I'm catching up with the last 12 hours on this thread. I think we are trying too hard and making privacy complicated. Anyone joining a jam session should not expect privacy EXCEPT when it is a closed jam session that they control. If you allow any visitors, then your jam session is not closed.

**Comment by ghost** [comment]
I guess the best mute detector is not hearing someone, or not seeing activity on their Vu meter. Maybe the client Vu meter could indicate a muted status if it was below a certain level for longer that a certain period.

The next thing would be Chat. Maybe Chat (with to:username) could be activated by right-clicking their slider and, selecting a Chat option, tell them to be online (unmute). Maybe the Chat client can highlight* a directed chat message; *or partition it. Chat could be used to send all kinds of status. Chat clients could also not display chats not addressed to them. (more for convenience, not absolute privacy, this is not any more or less private than now)

**Reply by DominikSchaller** [reply]
No, the idea is that there is an indication if somebody pressed Mute myself. The symbol should not be shown if somebody is just quiet for some time.

**Comment by gilgongo** [comment]
Hi all - I'm moving this to a Discussion until such time as we can agree a spec for this to add to the backlog or not (and it also seems like a consideration as part of the ongoing UI discussions too).

**Reply by henkdegroot** [reply]
I admit that showing the status of the Mute Myself is a very desired/elegant solution. With our band, it happens often someone has muted himself and forgot to unmute. Even though the red label at the top is shown.
Another approach, with less impact, could be to make a more obvious that you have muted yourself. 
![mute-myself](https://user-images.githubusercontent.com/13550012/111955242-8e0a8800-8ae9-11eb-977f-53857a23ccc1.PNG)

This can even be done as an optional settting, i.e. allow the user to select to use the existing small label or a big label and while we are working on that....perhaps add a setting to allow setting the default state to muted (I know the CLI already allows this but there are many persons who don't know how to use that).

I would even propose the label to be a button, so that when you click the big label it will unmute yourself as well.

**Reply by gilgongo** [reply]
That would certainly be hard to miss (assuming you're not red/green colourblind) :-) 

Do we have evidence that people are missing the current banner though?

**Reply by henkdegroot** [reply]
I did not realize this idea has already been discussed in the past....and indeed as stated in the past as well...this does not solve the actually issue that one forgot to unmute.

**Comment by houbsta** [comment]
I think whether or not a user's interface shows their mute status is irrelevant. I believe most cases are that the _user is now no longer looking_ at their screen and are in a "playing position" at their instrument. So they will just forget they have muted themselves. Other users in front of a screen with their jamulus client can tell them (audio) that they are muted, if they can see the status of other users' mute myself. Also - it's counter-intuitive that you "mute myself" but actually _you can hear yourself_. The button action is rather "mute my sound for others"...

As mentioned in other contexts (conf calls via Jitsi, Teams, Zoom, Meet...) people often have to be reminded they have muted themselves in spite of a big red icon on their screen. This is the same. If you mute yourself the status should display on that user for other clients. Making the UI super-ugly or super-obvious BIG RED MUTE is solving the wrong problem.

**Reply by henkdegroot** [reply]
My humble opinion....it is the user making this mistake in the first place....

Any way...you are very correct about the user no longer paying attention to the screen and they forgot to have muted themselves.
Guess Jamulus is on it's way to join Teams/Zooms/etc....in which the most common spoken phrase is: "I think you're on mute" :)

**Reply by gilgongo** [reply]
@houbsta An audio signal has been suggested in the past (eg a regular "ping" sound while you are muted)

@henkdegroot I'm not sure we have much evidence that mistakenly muting yourself is a sufficiently important problem in Jamulus compared to, say, not being able to sort out sound devices, feedback or keeping to The Golden Rule, but it's hard to say of course.

**Reply by DominikSchaller** [reply]
There will be (and are) situations in which people only want to listen. To them a regular "ping" would be disturbing. And forcing them to unmute themselves in order to have the ping disappear would be an unwanted behavior.

Also often Jamulus instances are used to monitor or record, for them an acustic signal would also not be good.

**Reply by dcorson-ticino-com** [reply]
There is also the, I think, most common situation where one uses mute myself, which is during tuning ones instrument.
Any "ping" tone would be very disturbing.

**Reply by gilgongo** [reply]
BTW the audio cue was discussed here but I see a window message was favoured instead:  https://github.com/jamulussoftware/jamulus/discussions/1040

**Reply by houbsta** [reply]
In my view the original request was for a third party to see if someone is muted; not for that user themselves to see (or hear) that they are. The use case is much more that a band rehearsing, or a livestream producer in particularly, has a way to know if someone is inaudible because they are actually muting themselves (easy to fix), or whether there is something else on their system (sound interface gain, hardware mixer mute, instrument based volume mute etc) they need to look for. This speeds up getting that person back in the jam :).

**Reply by ghost** [reply]
I have had to use the telephone to get someone* in the remote household to umute the sax player while he was playing live.
Sometimes it cannot be done in Zoom, or even Jamulus, so don't go extreme solving it with Jamulus. *Notice that sometimes you must alert someone else.

**Comment by jugo6** [comment]
I think the present red banner should me conspicuous enough for those who have jamulus in sight. If however people have something else fullscreenlike a sheet, chords or Zoom, then all signs are futile. Exept maybe for a popup in the taskbar or a change of icon in the taskbar, like a jamulus icon with a red cross or overlay. Sending popups in the taskbar when you play can also be annoying of course. I encourage choirmembers to keep on singing (muted) when I rehearse other parts.

**Comment by pcar75** [comment]
Making 'Mute Myself' visible to others !  Yes, absolutely .

I agree with Maallyn , DavidSavinkoff.  No need to be a sound engineer or a choir director to see that it would be quite helpful.

In any session (private or public server), it happens that someone gone to get a beer or whatever, forgets to unmute himself when coming back and thinks he is contributing to the music only to realize later that he/she wasn't "playing with the group".

In open session (public), it regularly happens that someone thinks of having audio problems (relating it with chat), delaying the session so others can try helping him (with audio or chat) through the usual suspects (hw, sw driver, settings, internet, etc.), only to discover that his/her client is in 'Mute Myself'.

It would also relax already in-session musicians to see that a new member or visitor will not hurt the session in progress as he/she has "Mute myself" prior to joining.  I've often noticed that the listener icon in the profile is used by contributing singers and (somewhat less) by other musicians.

It could appear to other clients with a specific led (yellow, orange, square or triangle i.e. different that the 'Muted' led and more easily identifiable for visually challenged users) ; it could appear as a red or grayed out or X marked channel fader or strip.

But again, it would be very helpful to many !

**Comment by gilgongo** [comment]
OK I think we probably have enough for a feature request ticket along these lines. I can put it on the backlog if the following seems a reasonable spec:

---

Currently, you cannot tell when somebody else has muted themselves using the "Mute myself" button.

**Pros/cons**

1. Knowing the "status" of another person is useful (various use cases cited).  
1. Jamulus is inconsistent with the "mute myself" convention/expectation in other apps such as Zoom.
1. There is a possible "privacy" advantage in not broadcasting the fact that you are muted, but other players can see you are on the server. 
1. Backward incompatibility means that it may not be universal for a very long time. But the main use cases appear to be for unregistered servers, so that may not be a big issue.

**Implementation**

Take the form of a "disabled" channel separate from the "muted" icon that current exists. 

We assume the indication will only show under the following conditions:

- only clients that have implemented the new feature
- only when connected to servers that have implemented the new feature
- and only for other connected clients that have implemented the new feature
- and not when muted outside Jamulus
- and not when simply not making any sound

The feature could also be subject to a preference toggle ("Show others I have muted myself") defaulting to ~~"on"~~ "off" to preserve the current behaviour by default

**Reply by rdica** [reply]
Defaulting to off? And possibly a cli arg for the server to enable it explicitly?

**Reply by gilgongo** [reply]
Yes, probably better defaulting to current behaviour seeing as it would be backwardly incompatible (although one of the scenarios is naive users not knowing that "mute myself" doesn't work like Zoom, so pros/cons)

What's the rationale for a switch on the server? That would then introduce a fifth factor...

**Reply by rdica** [reply]
I may have misunderstood the above summary in that I thought it meant changes to both client and server. As a server operator my preference is to not enable it regardless if client has it turned on. If it's client only, then default to off.

**Reply by gilgongo** [reply]
It means changes in the **code** to both client and server if the feature is to work at all, yes. 

But if both clients and server operators can choose whether or not to enable it, then that would give rise to the following possibilities _every time_ you notice a channel is not making any sound without indicating it's muted:

1. They have a client that is incapable of showing it.
1. The server is not capable of showing it (you could determine this by the version shown by explorer.jamulus.io)
1. They have not enabled showing it (and it may be off by default, so may not know about it).
1. The server operator has not enabled it (and may not know about it)
1. They just aren't making any sound right now.
1. They are muted outside of Jamulus.

**Reply by gilgongo** [reply]
> As a server operator my preference is to not enable it 

Incidentally, this is interesting as it touches on a previous conversation about the degree to which server operators should be given "powers" over clients. I think the outcome was that anything that affects the clients' functionality once connected should not be something server operators have control over. This is because it would tip the balance away from the currently egalitarian nature of Jamulus. But it's an interesting area of debate! :-)

**Comment by maallyn** [comment]
Excellent suggestion. I agree with rdica. 

Mark Allyn

**Reply by pcar75** [reply]
> As a server operator my preference is to not enable it regardless if client has it turned on (...)

A switch on the server enabling/disabling 'Mute Myself' display to session members when one has chosen to share such a condition would definitely  contradict that 'egalitarian nature' ...

> about the degree to which server operators should be given "powers" over client.

If it must, a client switch is quite sufficient.

**Reply by gilgongo** [reply]
> If it must, a client switch is quite sufficient.

Even if it wasn't a configurable option on the client, there might still be four other reasons why a muted channel might not show as muted (see https://github.com/jamulussoftware/jamulus/discussions/1025#discussioncomment-2397843)

This does make it a bit of a _strange_ feature to have. "If you see somebody has muted themselves, congratulations!" :-)
